github/github_restorer.py

```diff
@@ -0,0 +1,158 @@
+        prompt = " [Y/n] "
+    elif default == "no":
+        prompt = " [y/N] "
+    else:
+        raise ValueError("invalid default answer: '%s'" % default)
+
+    while True:
+        sys.stdout.write(question + prompt)
+        choice = raw_input().lower()
+        if default is not None and choice == '':
+            return valid[default]
+        elif choice in valid:
+            return valid[choice]
+        else:
+            sys.stdout.write("Please respond with 'yes' or 'no' "
+                             "(or 'y' or 'n').\n")
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('username', type=str,
+                        help='Github username')
+    parser.add_argument('repository', type=str,
+                        help='name of the repository to be restored')
+    parser.add_argument('-o', '--organization', dest='organization', type=str,
+                        default='ethz-asl',
+                        help='Github organization. Defaults to ethz-asl')
+    parser.add_argument('-f', '--force', dest='force',
+                        action='store_true',
+                        help='overwrite the Github repo if one exists already'
+                        ' (DANGEROUS). Off by default.')
+
+    args = parser.parse_args()
+
+    username = args.username
+    if "GITHUB_PASSWORD" in os.environ:
+        print "Taking password from GITHUB_PASSWORD environment variable."
+        password = os.environ.get("GITHUB_PASSWORD")
+    else:
+        print ("GITHUB_PASSWORD environment variable is not set, so will ask "
+              "for password.")
+        password = getpass.getpass("Enter password for %s: " % username)
+
+    try:
+        g = Github(username, password)
+
+        # Assuming for now script runs in same folder where repo is archived
+
+        # Check for backup and repo existence, wiki repo, and issues
+        if not os.path.exists(os.getcwd() + "/" + args.repository + ".git"):
+            print("Could not find repo directory in current working directory:"
+                  " " + args.repository + ".git")
+            sys.exit(1)
+
+        try:
+            g.get_organization(args.organization).get_repo(args.repository)
+            if args.force:
+                print "--force not implemented yet."
+                sys.exit(1)
+            else:
+                print ("Repository '" + args.repository + "' already exists on "
+                       "Github.")
+                sys.exit(1)
+        except GithubException as e:
+            pass
+
+        repo_has_wiki = os.path.exists(os.getcwd() + "/" + args.repository + \
+                        ".wiki.git")
+
+        # Restoring. First create repo
+        repo = g.get_organization(args.organization).create_repo(\
+               args.repository, private=True, has_wiki=repo_has_wiki)
+
+        # Push
+        ret = subprocess.call("cd " + args.repository + ".git && git push",
```

Comment:
@HannesSommer I believe so. At least --all is incompatible with --mirror (we clone repos with --mirror), so passing --all here results in an error.